arbitrum-docs/launch-orbit-chain/how-tos/customize-precompile.md

```diff
@@ -0,0 +1,150 @@
+---
+title: "How to customize your Orbit chain's precompiles"
+sidebar_label: "Customize your chain's precompiles"
+description: "Learn how (and when) to customize your Orbit chain's precompiles"
+author: jasonwan
+sidebar_position: 2
+---
+
+import PublicPreviewBannerPartial from '../partials/_orbit-public-preview-banner-partial.md';
+
+<PublicPreviewBannerPartial />
+
+import UnderConstructionPartial from '../../partials/_under-construction-banner-partial.md';
+
+<UnderConstructionPartial />
+
+There are two ways to customize your chain's precompiles:
+
+1.  Add new methods to to an existing [precompile](https://github.com/OffchainLabs/nitro-contracts/tree/97cfbe00ff0eea4d7f5f5f3afb01598c19ddabc4/src/precompiles).
+2.  Create a new precompile.
+
+### Prerequisites
+
+Clone the Nitro repository before you begin:
+
+```shell
+git clone <https://github.com/OffchainLabs/nitro.git>
+cd nitro
+git submodule update --init --recursive --force
+```
+
+## Option 1: Add new methods to an existing precompile
+
+Using your favorite code editor, open an existing precompile from the [precompiles/](https://github.com/OffchainLabs/nitro/tree/master/precompiles) directory. We'll use `ArbSys.sol` as an example. Open the the corresponding `go` file (`ArbSys.go`) and add a simple `SayHi` method:
+
+```go
+func (con *ArbSys) SayHi(c ctx, evm mech) (string, error) {
+	return "hi", nil
+}
+```
+
+Then, open the corresponding Solidity file (`ArbSys.sol`) from the [contracts/src/precompiles/](https://github.com/OffchainLabs/nitro-contracts/tree/97cfbe00ff0eea4d7f5f5f3afb01598c19ddabc4/src/precompiles) directory and add the required interface. Ensure that the method name on the interface matches the name of the function you introduced in the previous step, `camelCased`:
+
+```solidity
+function sayHi() external view returns(string memory);
+```
+
+Next, build Nitro by following the instructions in [How to build Nitro locally](/node-running/how-tos/build-nitro-locally). Note that if you've already built the Docker image, you still need run the last step to rebuild.
+
+Run Nitro by following the instructions in [How to run a full node](/node-running/how-tos/running-a-full-node#putting-it-all-together).
```

Comment:
Idea: as this is specific for orbit chains, we could maybe add the command here and a note below saying something like "to know more about all available parameters read [How to run a full node](/node-running/how-tos/running-a-full-node#putting-it-all-together)".

Reply:
Yes, we need other command line to run orbit, how do you think if we point them to the orbit setup script then use that docker-compose file and tell user to change the image name.

Reply:
Oh, that's right. That's better, yeah. Show them where to change the image name in the docker-compose file.

Reply:
Love it, definitely aligned with the general principle of "avoid forcing readers to go over there and come back".

I probably wouldn't block this PR on its implementation, but we can optimize via another PR, eg by partializing the command and including in both places.